test/Microsoft.ML.Core.Tests/UnitTests/CoreBaseTestClass.cs

```diff
@@ -85,75 +76,10 @@ protected Func<bool> GetComparerVec<T>(DataViewRow r1, DataViewRow r2, int col,
                 {
                     g1(ref v1);
                     g2(ref v2);
-                    return CompareVec<T>(in v1, in v2, size, fn);
+                    return CommonUtilities.CompareVec<T>(in v1, in v2, size, fn);
                 };
         }
 
-        protected bool CompareVec<T>(in VBuffer<T> v1, in VBuffer<T> v2, int size, Func<T, T, bool> fn)
-        {
-            return CompareVec(in v1, in v2, size, (i, x, y) => fn(x, y));
-        }
-
-        protected bool CompareVec<T>(in VBuffer<T> v1, in VBuffer<T> v2, int size, Func<int, T, T, bool> fn)
-        {
-            Contracts.Assert(size == 0 || v1.Length == size);
-            Contracts.Assert(size == 0 || v2.Length == size);
-            Contracts.Assert(v1.Length == v2.Length);
-
-            var v1Values = v1.GetValues();
-            var v2Values = v2.GetValues();
-
-            if (v1.IsDense && v2.IsDense)
-            {
-                for (int i = 0; i < v1.Length; i++)
-                {
-                    var x1 = v1Values[i];
-                    var x2 = v2Values[i];
-                    if (!fn(i, x1, x2))
-                        return false;
-                }
-                return true;
-            }
-
-            Contracts.Assert(!v1.IsDense || !v2.IsDense);
-            int iiv1 = 0;
-            int iiv2 = 0;
-            var v1Indices = v1.GetIndices();
-            var v2Indices = v2.GetIndices();
-            for (; ; )
-            {
-                int iv1 = v1.IsDense ? iiv1 : iiv1 < v2Indices.Length ? v1Indices[iiv1] : v1.Length;
```

Comment:
>v2Indices [](start = 53, length = 9)

Seems like a bug to me, should be v1Indices

Reply:
Good catch! Yes, it does appear to be a bug. Can you please check the callers who are using this function and how those tests are passing?

---
In reply to: [336252686](https://github.com/dotnet/machinelearning/pull/4346#discussion_r336252686) [](ancestors = 336252686)

Reply:
TestEntryPoints class is using this method, these tests will pass as long as v1 and v2 has same indices length, this happens to be true in these tests

---
In reply to: [336584593](https://github.com/dotnet/machinelearning/pull/4346#discussion_r336584593) [](ancestors = 336584593,336252686)